### PR TITLE
Update on hover CSS for AppTile component

### PR DESCRIPTION
**Pull request by stephanieyoshimoto-okta**
[OKTA-811761](https://oktainc.atlassian.net/browse/OKTA-811761)

## Summary

- Remove background color for on hover behavior
- Add custom styling for AppTile (won't affect other components that use MUI's Card)
- Add grey background to storybook

## Testing & Screenshots

- [x] I have confirmed this change with my designer and the Odyssey Design Team.

### CODE REVIEW

File: packages/odyssey-react-mui/src/labs/AppTile.tsx [hunk 1]
```diff
@@ -169,7 +169,15 @@ const AppTile = ({
   );
 
   return (
-    <MuiCard className="isClickable">
+    <MuiCard
+      className="isClickable"
```
Comment (bryancunningham-okta) [inline packages/odyssey-react-mui/src/labs/AppTile.tsx:173]: @stephanieyoshimoto-okta We can remove the className here. Can just do
```js
"&:hover": {
  "box-shadow": odysseyDesignTokens.DepthMedium,
},
```

Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/labs/AppTile.tsx:173]: Or, better yet, we could move all this into a styled component. That is the method we've been leaning towards when we can. Let me know if you have questions about what that looks like and/or have any issues converting this to a styled component

File: packages/odyssey-react-mui/src/labs/AppTile.tsx [hunk 2]
```diff
@@ -169,7 +169,15 @@ const AppTile = ({
...
+      sx={{
+        boxShadow: "none",
+        "&.isClickable:hover": {
+          "box-shadow": odysseyDesignTokens.DepthMedium,
```
Comment (bryancunningham-okta) [inline packages/odyssey-react-mui/src/labs/AppTile.tsx:177]: Should this be `DepthHigh`? Like the other cards?

Comment (stephanieyoshimoto-okta) [reply packages/odyssey-react-mui/src/labs/AppTile.tsx:177]: we're trying to make this unique to the `AppTile`, not all `MuiCard`'s. `MuiCard` goes from medium -> high, and we want `AppTile` to go from nothing -> medium

Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/labs/AppTile.tsx:177]: All good. Assuming design was ok with this then?

Comment (stephanieyoshimoto-okta) [reply packages/odyssey-react-mui/src/labs/AppTile.tsx:177]: yes https://okta.slack.com/archives/C7T2H3KNJ/p1727371369514169?thread_ts=1727303816.749929&cid=C7T2H3KNJ

File: packages/odyssey-react-mui/src/labs/AppTile.tsx [hunk 3]
```diff
@@ -114,6 +114,19 @@ const ChildrenContainer = styled("div", {
   },
 }));
 
+const StyledMuiCard = styled(MuiCard, {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{ odysseyDesignTokens: DesignTokens }>(({ odysseyDesignTokens }) => ({
+  boxShadow: "none",
+  "&::after": {
+    opacity: 0,
+    "box-shadow": odysseyDesignTokens.DepthMedium,
```
Comment (bryancunningham-okta) [inline packages/odyssey-react-mui/src/labs/AppTile.tsx:123]: We'll need to transition the opacity here as well

File: packages/odyssey-react-mui/src/theme/components.tsx [hunk 4]
```diff
@@ -856,7 +856,7 @@ export const components = ({
             paddingLeft: odysseyTokens.Spacing4,
           },
 
-          "&.isClickable:hover": {
+          "&:hover": {
```
Comment (bryancunningham-okta) [inline packages/odyssey-react-mui/src/theme/components.tsx:859]: This we probably need to keep. You didn't add this did you? I think removing this here breaks other implementations

File: packages/odyssey-react-mui/src/labs/AppTile.tsx (hunk 3 above)
Comment (stephanieyoshimoto-okta) [reply packages/odyssey-react-mui/src/labs/AppTile.tsx:123]: there's a transition already in the overrides for `MuiCard`. do we need to override that?

Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/labs/AppTile.tsx:123]: Hmm, I see that. I don't think we really want to transition "all". I'd assume that was put in place for `box-shadow` only. But, we'd have to ask. @jordankoschei-okta Any additional context here? 

Ideally the `box-shadow` transition goes away and we are only transitioning `opacity`. That would involve moving the `MuiCard` to this same pseudo element solution though

**Review by jordankoschei-okta — APPROVED** [review]

**Pull request merged**